perllib/FixMyStreet/Cobrand/Bexley/Waste.pm

```diff
@@ -403,24 +403,20 @@ sub bin_services_for_address {
             $filtered_service->{schedule} = 'Weekly';
         }
 
-        my $report_details = $property->{open_reports}{missed}
-            { $filtered_service->{service_id} };
-
-        if ($report_details) {
-            $filtered_service->{report_details} = $report_details;
-            $filtered_service->{report_open} = $report_details->{open};
-        } else {
-            $filtered_service->{report_open} = 0;
-        }
-
-        my $request_details = $property->{open_reports}{request}
-            { $filtered_service->{service_id} };
-
-        if ($request_details) {
-            $filtered_service->{request_details} = $request_details;
-            $filtered_service->{requests_open} = $request_details->{open};
-        } else {
-            $filtered_service->{requests_open} = 0;
+        foreach (
+            { type => 'missed', open => 'report_open', details => 'report_details' },
+            { type => 'delivery', open => 'delivery_open', details => 'delivery_details' },
+            { type => 'removal', open => 'removal_open', details => 'removal_details' },
+        ) {
```

Comment:
Why do we have subtypes again? As opposed to just three types for the different sizes? :)

I don't think we have the subtypes 'available' at this point; we haven't called set_request_containers yet either.

Reply:
I've added a reverse - I don't *think* it's needed, because I could use `_containers()` (or the equivalent) that the remove form uses for the ID, but that appears to be missing some entries so added this for now, can chat about it tomorrow.

Reply:
We have the subtypes because it's three sizes under one umbrella/name of 'Green Wheelie Bin' etc. I don't know if there would have been a better way of handling it in retrospect.